language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/MakeAnnotationDeclConstantCodeAction.java

```diff
@@ -0,0 +1,110 @@
+import org.ballerinalang.langserver.commons.CodeActionContext;
+import org.ballerinalang.langserver.commons.codeaction.spi.DiagBasedPositionDetails;
+import org.eclipse.lsp4j.CodeAction;
+import org.eclipse.lsp4j.CodeActionKind;
+import org.eclipse.lsp4j.Position;
+import org.eclipse.lsp4j.Range;
+import org.eclipse.lsp4j.TextEdit;
+
+import java.net.URI;
+import java.nio.file.Path;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * Code Action to make annotation declaration constant.
+ *
+ * @since 2201.1.1
+ */
+@JavaSPIService("org.ballerinalang.langserver.commons.codeaction.spi.LSCodeActionProvider")
+public class MakeAnnotationDeclConstantCodeAction extends AbstractCodeActionProvider {
+    public static final String NAME = "Make Annotation Declaration Constant";
+    public static final String DIAGNOSTIC_CODE = "BCE2638";
+
+    @Override
+    public boolean validate(Diagnostic diagnostic,
+                            DiagBasedPositionDetails positionDetails,
+                            CodeActionContext context) {
+        if (!DIAGNOSTIC_CODE.equals(diagnostic.diagnosticInfo().code()) || context.currentSyntaxTree().isEmpty() ||
+                context.currentSemanticModel().isEmpty()) {
+            return false;
+        }
+
+        SyntaxTree syntaxTree = context.currentSyntaxTree().orElseThrow();
+        NonTerminalNode matchedNode = CommonUtil.findNode(new Range(context.cursorPosition(),
+                context.cursorPosition()), syntaxTree);
+        return CodeActionNodeValidator.validate(matchedNode);
+    }
+    
+    @Override
+    public List<CodeAction> getDiagBasedCodeActions(Diagnostic diagnostic,
+                                                    DiagBasedPositionDetails positionDetails,
+                                                    CodeActionContext context) {
+        Optional<Pair<Symbol, Path>> symbolAndPath = CodeActionUtil.findSymbolAndPathForDiagnosticRange(diagnostic, 
+                context);
+        if (symbolAndPath.isEmpty()) {
+           return Collections.emptyList();
+        } 
+        Symbol symbol = symbolAndPath.get().getLeft();
+        Path filePath = symbolAndPath.get().getRight();
+        
+        URI uri = filePath.toUri();
+        Optional<NonTerminalNode> node = CommonUtil.findNode(symbol,
+                context.workspace().syntaxTree(filePath).get());
+        if (node.isEmpty()) {
+            return Collections.emptyList();
+        }
+
+        AnnotationDeclarationNode annotationDeclarationNode = (AnnotationDeclarationNode) node.get();
+        Position position = CommonUtil.toPosition(annotationDeclarationNode.annotationKeyword().lineRange()
+                .startLine());
+
+        Range range = new Range(position, position);
+        String editText = SyntaxKind.CONST_KEYWORD.stringValue() + " ";
+        TextEdit textEdit = new TextEdit(range, editText);
+        List<TextEdit> editList = List.of(textEdit);
```

Comment:
Shall we execute the `Range` and `TextEdit` in `line 95`? since we're not using them in any other place?